setup/generate.ps1

```diff
@@ -0,0 +1,142 @@
+    When the MSI installs it will install the cmdlets to the same location the gallery would, and PowerShellGet treats them 
+    as if they were installed from the gallery
+.PARAMETER Version
+    The version number for the generated MSI.
+.PARAMETER Force
+    Forces a fresh installation of the Azure and AzureRm cmdlets from the gallery
+.PARAMETER noBuildNumber
+    Prevent a build number from being tacked on the end of the version number.
+.PARAMETER repository
+    Set the repository to pull packages from.
+#>
+
+param(
+    [Parameter(HelpMessage="The version number for the generated MSI.")]
+    [string]$version,
+
+    [Parameter(HelpMessage="Forces a fresh installation of the Azure and AzureRm cmdlets from the gallery")]
+    [Switch]$force,
+    
+    [Parameter(HelpMessage="Prevent a build number from being tacked on the end of the version number.")]
+    [Switch]$noBuildNumber,
+
+    [Parameter(HelpMessage="Set the repository to pull packages from.")]
+    [string]$repository="PSGallery"
+    )
+
+if( (-not (get-command -ea 0 light)) -or (-not (get-command -ea 0 heat)) -or (-not(get-command -ea 0  candle)) ) {
+    write-host -fore Red  "This script requires WiX Toolset in the path. [exiting]"
+    return ;
+}
+
+# variables 
+
+# the build number is the number of commits in this branch. 
+
+if( -not $noBuildNumber ) {
+    # useful for an ever-increasing number that can be tracked to a commit.
+    $buildNumber = git rev-list --parents HEAD --count --full-history
+
+    # tack the build number onto the version
+    $version = "$version.$buildNumber"
+}
+
+# output filename (plus '-$version-$arch.msi' )
+$outputName ="Azure-Cmdlets"
+
+# generate the product name from the current month/year.
+$productName = "Microsoft Azure PowerShell - $((Get-Culture).DateTimeFormat.GetMonthName((get-date).month)) $((get-date).year)"
+
+# where to put temp files
+$tmp = Join-Path $env:temp azure-cmdlets-tmp
+
+# where we're going to install the modules to 
+$modulesDir = "$tmp\modules"
+
+# archetectures supported
+$archs = @('x86','x64')
+
+# cleanup first
+if( $force  ) {
+    Write-Host -fore yellow "Forcing clean install"
+    $shhh = (cmd.exe /c rmdir /s /q $tmp)
+}
+
+$shhh = mkdir -ea 0 "$tmp"
+erase -ea 0 "$tmp/*.wixobj"
+erase -ea 0 "$tmp/*.wxi"
+
+erase -ea 0 "$PSSCRIPTROOT/*.wixpdb"
+erase -ea 0 "$PSSCRIPTROOT/*.msi"
+
+# install modules locally.
+if ( -not (test-path $modulesDir))  {
+    Write-Host -fore green "Installing Modules..."
+    # create a folder for everything
+    $shh= mkdir -ea 0 $modulesDir 
+
+    # First install the modules locally
+    save-module azure -path $modulesDir -Repository $repository
+    save-module azurerm -path $modulesDir -Repository $repository
+
+    Write-Host -fore green "Tweaking Modules"
+    cmd /c dir /a/s/b "$modulesDir\psgetmoduleinfo.xml" |% {
+        Write-Host -fore Gray " - Patching $_"
+        (gc $_ -raw ) -replace ".*<S N=.InstalledLocation.*S>",""  | Set-Content $_
```

Comment:
This is going to retain the information about which gallery the module came from, right?  Ideally we would set this to PSGallery, or allow the script writer to override, so that we could build official modules from build drops.

Reply:
It does retain the upstream information, I only remove the `InstalledLocation` from the file (which WMF5.0 didn't support anyway, and the behavior is to assume the folder it's installed in.)

Reply:
I did not look at how to alter the feed information -- I think it's a smidgen dangerous, IIRC, it records the 'name' of the feed, which would only be salient for the default feed AFAIK.

Reply:
@fearthecowboy Can you sync with the Gallery team about this?

Reply:
@MiYanni  What specifically do you want me to 'sync with' them on.  The Gallery that's used is stored in   `RepositorySourceLocation` and `Repository` in the psgetmoduleinfo.xml. If the location and name don't match one the user has, it won't be able update. The only source you can bank on is the 'default' gallery.

Reply:
Which one could overwrite, but given the fragility of the `psgetmoduleinfo.xml` file, I'd highly recommend not tweaking that as much as possible.

Reply:
The only reason I can get away with it for InstalledLocation is that the original PSGet didn't store that, so they have a fall-back means to calculate the location based on where it was found